### ISSUE TITLE
[Feature Request] handle raw representation of Any message in fromJson

### ISSUE THREAD

**Issue by jackielii** [bufbuild/protobuf-es#921]
## The problem

The protobuf spec defines the JSON mapping of Any type to have:

```
{"@type": xxx, "value": yyy}
```

In our implementation, we did it correctly by checking if `@type` exists: https://github.com/bufbuild/protobuf-es/blob/2966b9cba6713fada2fca5670cbe14ee4798f945/packages/protobuf/src/google/protobuf/any_pb.ts#L188-L194

However, in practice, I often serialise `Any` type using `json` instead of `protojson`, so the Any message will be serialised as `{ "type_url": "...", value: "..."}`.

In addition, the Any class converts the deserialised form back into the binary form within `fromJson`: https://github.com/bufbuild/protobuf-es/blob/2966b9cba6713fada2fca5670cbe14ee4798f945/packages/protobuf/src/google/protobuf/any_pb.ts#L207 This is unnecessary if I already have the Any type in its "raw" form. Plus, I don't have to provide the type registry until I want to unpack it.

To sum it up:

1. `Any.fromJson` should deal with messages with `{ "type_url": "...", value: "..." }`.
2. this saves the serialisation
3. user doesn't need to provide the type registry

## Proposal

Add a case where `@type` is absent but `type_url` is present to construct the `Any` message

## Work arounds considered

1. I considered using binary serialisation, but I lose readability
2. I provided type registry and used protojson to serialise it, but I had to provide the type registry which is not known at the top level. It's only available when the message is passed to the rendering (last step) where each component deals with it's type.

**Comment by timostamm** [comment]
> in practice, I often serialise Any type using `json` instead of `protojson`

The downside is that `json` does not serialize to the standard protobuf JSON format, so you end up with incompatibilities with other implementations. The documentation on `protojson` states:

> encoding/json [...] does not operate correctly on protocol buffer messages.

I don't think it's practical to update all other implementations to support this format.  `Any` isn't perfect, but it's fantastic to have a reliable standard.

If you don't want the overhead of `Any` in JSON, can't you simply define your own `message Any { string type = 1; bytes value = 2;}`?

**Comment by jackielii** [comment]
@timostamm thanks for responding, you're probably right that there is other things that makes json.Marshal incompatible with json. I'll close this one as I do have simple workarounds with using binary serialisation. In addition, I tried using patch-package to patch the `@bufbuild/protobuf`, which also works for now. I can report back anything else if I go down this route.

Thanks again.
